src/main/java/uk/gov/hmcts/reform/sscs/services/CaseLoaderService.java

```diff
@@ -76,32 +73,44 @@ public void process() {
         }
     }
 
-    private void processDelta(IdamTokens idamTokens, Gaps2File file) {
+    private void processDelta(Gaps2File file) {
         List<SscsCaseData> cases = transformService.transform(sftpSshService.readExtractFile(file));
         log.info("*** case-loader *** file transformed to {} Cases successfully", cases.size());
+        int counter = 0;
+        IdamTokens idamTokens = idamService.getIdamTokens();
```

Comment:
I think first time we should create the idam token by using IdamTokens idamTokens = IdamTokens.builder()	
             .idamOauth2Token(oauth2Token)	
             .serviceAuthorization(idamService.generateServiceAuthorization())	
             .userId(idamService.getUserId(oauth2Token))	
             .build();

Reply:
don't see any reason for that @srochani .  Could you elaborate, please?

Reply:
I don't know much about the idam, but looking at the code, earlier we used to generate idam token by passing the userId, but now in new code we are not using the userId.
As per my understanding, we are going to request the tokens after 100 cases but before that we have to create/generate the token based on userId then refresh the token after 100 cases.
Please correct me if I'm wrong

Reply:
Sunnil, the userId is inferred from the token, actually. the method getIdamTokens takes care of generating the token and retrieve the userId correctly for that token. This is the same that we had before. The only difference now is that after a certain number of processed cases we update the oauth2 token, only this token. 
It's all fine here